### ISSUE TITLE
publish minified content to npm?

### ISSUE THREAD

**Issue by pdehaan** [projectfluent/fluent-web#7]
We're using **fluent-web** on the upcoming TestPilot site, but I noticed that when we copy the assets over from ./node_modules/ the contents aren't minified.

https://github.com/mozilla/testpilot/blob/852c88f9664f6d5ddc415fd6f5a86cd751243d64/bin/build-static-dist.js#L12-L13

Other files you could potentially not publish to the npm bundle are:

- .eslintrc.js
- CHANGELOG.md
- common.mk
- ./examples/*
- rollup_config.js

```sh
$ tree ./node_modules/fluent-web -a

./node_modules/fluent-web
├── .eslintrc.js [50B]
├── CHANGELOG.md [328B]
├── README.md [2K]
├── common.mk [1.4K]
├── examples/
│   ├── localization/
│   │   ├── en-US/
│   │   │   ├── fluent-widget.ftl
│   │   │   └── main.ftl
│   │   └── pl/
│   │       ├── fluent-widget.ftl
│   │       └── main.ftl
│   ├── simple.html
│   └── webcomponent.html
├── fluent-web.js [79K]
├── package.json [2.2K]
├── rollup_config.js [222B]
└── src/
    └── index.js [2K]

5 directories, 14 files
```

**Comment by pdehaan** [comment]
Not sure what the best ES6-style minifier is these days, but I was playing w/ **babel-minify** and it looks like it _may_ be able to reduce the bundle size by about 56K (from 79K down to 23K):

```sh
$ npx babel-minify ./node_modules/fluent-web/fluent-web.js > ./node_modules/fluent-web/fluent-web.min.js

$ ls -lash ./node_modules/fluent-web/fl*
    79K 26 Oct  1985 ./node_modules/fluent-web/fluent-web.js
    23K 21 Jan 17:03 ./node_modules/fluent-web/fluent-web.min.js
```

**Comment by zbraniecki** [comment]
@pdehaan - Would you be willing to provide a PR?

**Comment by pdehaan** [comment]
@zbraniecki I could try.

How are you using rollup? I could possibly try adding [**rollup-plugin-babel-minify**](https://github.com/Comandeer/rollup-plugin-babel-minify).

I tried the following, but just got errors because it doesn't look like rollup config has an input or output:

```sh
$ npx rollup -c ./rollup_config.js

 → stdout...
[!] Error: You must supply options.input to rollup
Error: You must supply options.input to rollup
    at Promise.all.then (/Users/pdehaan/dev/github/pdehaan/tmp/node_modules/@fluent/web/node_modules/rollup/dist/rollup.js:16759:23)
```

I also didn't see any mentions to rollup in the package.json scripts or common.mk.

---

#### Unrelated thoughts

What are the minimum files that need to be published with this module?

Here's the latest file tree vs `@fluent/web` v0.2.0:

```sh
tree node_modules/@fluent/web

node_modules/@fluent/web
├── CHANGELOG.md
├── README.md
├── common.mk
├── eslint_src.json
├── eslint_test.json
├── examples
│   ├── localization
│   │   ├── en-US
│   │   │   ├── fluent-widget.ftl
│   │   │   └── main.ftl
│   │   └── pl
│   │       ├── fluent-widget.ftl
│   │       └── main.ftl
│   ├── simple.html
│   └── webcomponent.html
├── fluent-web.js
├── package.json
├── rollup_config.js
└── src
    └── index.js

5 directories, 15 files
```

```sh
$ npm pack --dry-run

npm notice 📦  @fluent/web@0.2.0
npm notice === Tarball Contents ===
npm notice 2.3kB  package.json
npm notice 50B    .eslintrc.js
npm notice 418B   CHANGELOG.md
npm notice 1.0kB  common.mk
npm notice 3.1kB  eslint_src.json
npm notice 387B   eslint_test.json
npm notice 82.7kB fluent-web.js
npm notice 2.0kB  README.md
npm notice 202B   rollup_config.js
npm notice 99B    examples/localization/en-US/fluent-widget.ftl
npm notice 149B   examples/localization/en-US/main.ftl
npm notice 104B   examples/localization/pl/fluent-widget.ftl
npm notice 117B   examples/localization/pl/main.ftl
npm notice 634B   examples/simple.html
npm notice 2.2kB  examples/webcomponent.html
npm notice 2.1kB  src/index.js

npm notice === Tarball Details ===
npm notice name:          @fluent/web
npm notice version:       0.2.0
npm notice filename:      fluent-web-0.2.0.tgz
npm notice package size:  27.5 kB
npm notice unpacked size: 97.6 kB
npm notice shasum:        5da502983f7ee0dd5932d3ae44cbc413d3edb3e1
npm notice integrity:     sha512-8Q2tdmtOGnB4g[...]48Rt3KY7fT+iA==
npm notice total files:   16
npm notice
fluent-web-0.2.0.tgz
```

I'm thinking you may only need:
- fluent-web.js (82.7kB)
- package.json (2.3kB)
- src/index.js (2.1kB)

If we add this to package.json, it shaves off a quick 3-8 kB:

```diff
+  "files": [
+    "fluent-web.js",
+    "src/index.js"
+  ],
```

```sh
$ npm pack --dry-run

npm notice 📦  @fluent/web@0.2.0
npm notice === Tarball Contents ===
npm notice 2.3kB  package.json
npm notice 418B   CHANGELOG.md
npm notice 82.7kB fluent-web.js
npm notice 2.0kB  README.md
npm notice 2.1kB  src/index.js

npm notice === Tarball Details ===
npm notice name:          @fluent/web
npm notice version:       0.2.0
npm notice filename:      fluent-web-0.2.0.tgz
npm notice package size:  24.7 kB
npm notice unpacked size: 89.6 kB
npm notice shasum:        e61a952244a99e4896d8f8f43b20fb6fbbf8a772
npm notice integrity:     sha512-IHqVRSVhkoVJX[...]7zRs6W2iQKdcA==
npm notice total files:   5
npm notice
fluent-web-0.2.0.tgz
```

**Comment by zbraniecki** [comment]
Released 0.2.1 with the change! https://www.npmjs.com/package/@fluent/web

Thank you!
